src/service/agent.js

```diff
@@ -177,6 +178,16 @@ async function executeJob(jobInfo, keepFiles) {
     downloader.logger = jLogger;
     await downloader.download(tmpDirPath);
 
+    // if the extraFiles is not provided, the agent will work as normal flow
+    if (extraFiles !== undefined && extraFiles !== null) {
+      if (extraFiles.iterator !== undefined) {
+        for (const extraFile of extraFiles) {
+          // eslint-disable-next-line no-await-in-loop
+          await file.downloadExtraFileFromTestOps(extraFile, tmpDirPath, jLogger);
```

Comment:
Can we use await Promise.al(..)l for this case ?

Reply:
I think that we want to make sure that agent get the extraFile in order ?

Reply:
No. the order doesn't matter

Reply:
Update use Promise.all